### PR TITLE
feat: select_for_update in bulk_select_model_dicts

### PR DESCRIPTION
**Pull request by taylor-cedar**
## Summary
Ticket: https://cedar-jira.atlassian.net/browse/CED-67107

This adds the ability to add `FOR UPDATE` to the `bulk_select_model_dicts` function. This is useful for locking to rows when performing an update

##### How can reviewers test your change manually?
Check changed tests. Call the function with the new argument and check the rows are locked
##### What automated tests did you add? If none, please state why.
Added automated tests to check for lock

## Security Impact
Please answer the questions below about your PR. 
If yes, please explain and add `cedar-team/security` as reviewers.

##### Add/modify authentication, authorization, encryption, or HTTP headers?
No
##### Add/modify any PII or PHI data fields?
No
##### Add new third parties the app interacts with (e.g., Nordis/Stripe)?
No
##### Ask patients to enter a new kind of data, (e.g. insurance capture)?
No
##### Impact any of the other [Security Engineering Practices](https://careportal.atlassian.net/wiki/spaces/SEC/pages/1192690295/Security+Engineering+Practices)?
_Protect sensitive data, Validate your inputs, Know your user, Establish trust boundaries, Keep good records, Expect vulnerabilities, Follow least privilege_
No

### CODE REVIEW

File: tests/test_bulk_select_model_dicts_no_transaction.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                model_class=TestComplexModel,
+                filter_field_names=["integer_field"],
+                select_field_names=["json_field", "id"],
+                filter_data=[(saved_model1.integer_field,), (saved_model2.integer_field,)],
+                select_for_update=True
+            )
+            self.assertEqual(len(result_dicts), 2)
+
+            # Should be 1 because the 2 rows are locked above
+            self.assertEqual(get_unlocked_rows(), 1)
```
Comment (sbernstein-cedar) [inline tests/test_bulk_select_model_dicts_no_transaction.py:64]: isn't this only true because of `transaction.atomic()`? how does `select_for_update` make a difference?

Comment (taylor-cedar) [reply tests/test_bulk_select_model_dicts_no_transaction.py:64]: `transaction.atomic()` just creates a transaction. It doesn't lock any rows. We need to add `FOR UPDATE` to lock the rows.

https://www.cockroachlabs.com/blog/select-for-update/

Comment (sbernstein-cedar) [reply tests/test_bulk_select_model_dicts_no_transaction.py:64]: ah gotcha, thank you!

**Review by sbernstein-cedar — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [x] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)
